src/library_promise.js

```diff
@@ -137,13 +138,14 @@ mergeInto(LibraryManager.library, {
 #if RUNTIME_DEBUG
     dbg('emscripten_promise_then: ' + id);
 #endif
+    {{{ runtimeKeepalivePop() }}};
```

Comment:
Hmm, I think maybe you moved the pop instead of the push? We should do a `runtimeKeepalivePush` here and do the `runtimeKeepalivePop` in the callback.